packages/frontend/src/pages/NewListing.tsx

```diff
@@ -0,0 +1,445 @@
+        }
+        {currentStep < FormSteps.length &&
+          <button
+            type="button"
+            className='px-2 py-1 justify-self-end'
+            onClick={async () => {
+              const isValid = await trigger(FormSteps[currentStep - 1].fields);
+              if (isValid) {
+                changeStep(FormSteps[currentStep]);
+              }
+            }}
+          >
+            Next step
+          </button>
+        }
+        {currentStep === FormSteps.length &&
+          <button className='px-2 py-1 bg-blue-600 hover:bg-blue-400 text-background' type="submit">Publish</button>
+        }
+      </section>
+    </section>
+  )
+}
+
+const NewListingPage = () => {
+  const { calcScoreFromUserData, createNewListing } = useTrustlist()
+  const user = useContext(User) // TODO: This should be a hook
+  const navigate = useNavigate()
+  const [step, changeStep] = useState<FormStep>(FormSteps[0])
+  const [trustScoresFromData, setTrustScoresFromData] = useState({ ...trustScores }); // Make copy we can use
+  const currentStepNumber = FormSteps.findIndex(fs => fs.id === step.id) + 1
+
+  const listForm = useForm({
+    resolver: zodResolver(NewListingResponseSchema),
+    defaultValues: initialFormState.fields,
+  });
+  const onFormError = (errors: FieldErrors) => console.error({ errors })
+
+  const trustScoreKeys = Object.keys(TrustScoreKeyEnum) as (keyof typeof TrustScoreKeyEnum)[]
+
+  const updateScores = useCallback(() => {
+    if (user.provableData.length === 0) return;
+    for (let i = 0; i < 4; i++) {
+      let cumulativeScore = calcScoreFromUserData(Number(user.provableData[i]))
+      setTrustScoresFromData((prevData) => {
+        return {
+          ...prevData,
+          [TrustScoreKeyEnum[trustScoreKeys[i]]]: {
+            ...prevData[TrustScoreKeyEnum[trustScoreKeys[i]]],
+            score: cumulativeScore
+          }
+        }
+      })
+    }
+  }, [user.provableData]);
+
+  useEffect(() => {
+    updateScores();
+  }, [])
+
+  const getEpochAndKey = async () => {
+    const { userState } = user
+    if (!userState) throw new Error('Please join as a member and try again');
+    const currentEpoch = userState.sync.calcCurrentEpoch()
+    if (currentEpoch !== (await userState.latestTransitionedEpoch())) {
+      // transition user to the current epoch if they're not on it
+      // TODO: Set loading state
+      try {
+        console.log('transitioning...') // TODO: Add loading state on UI
+        await user.transitionToCurrentEpoch()
+        updateScores();
+      } catch (error) {
+        throw new Error("Failed to transition to the new epoch");
+      }
+    }
+
+    const epkNonce = Math.floor(Math.random() * 3)// randomly choose between 2 and 0
+    const posterId = user.epochKey(epkNonce) 
+
+    return { currentEpoch: currentEpoch, userEpochKey: posterId, nonce: epkNonce }
+  }
+
+  const publishPost = async (data: ListingFormValues) => {
```

Comment:
@CJ-Rose This function takes the `revealTrustScores` from the toggle inputs and churns out the scores (or thats what its supposed to do lol)